what4/src/What4/Protocol/SMTLib2.hs

```diff
@@ -938,6 +977,20 @@ instance SMTLib2Tweaks a => SMTReadWriter (Writer a) where
         cmd = getUnsatCoreCommand p
     in getLimitedSolverResponse "unsat core" unsatCoreRsp s cmd
 
+  smtAbductResult p s nm t =
+    let abductRsp = \case
+          AckSuccessSExp (SApp (_ : _ : _ : _ : abduct)) -> Just (tail $ init $ sExpToString (SApp abduct))
+          _ -> Nothing
+        cmd = getAbductCommand p nm t
+    in getLimitedSolverResponse "get abduct" abductRsp s cmd
+
+  smtAbductNextResult p s =
+    let abductRsp = \case
+          AckSuccessSExp (SApp (_ : _ : _ : _ : abduct)) -> Just (tail $ init $ sExpToString (SApp abduct))
```

Comment:
Another occurrence of the tricky `tail $ init` pattern. It would be worth factoring this out into its own function.